File: 58. Length of Last Word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        
        count = 0
        ans = []
        for x in range(len(s)):
            if s[x]!=" ":
                count+=1
            elif s[x] == " ":
                ans.append(count)
                count = 0
                
                
        ans.append(count)
        #print(ans)
        if len(ans) ==1:
            return count
        elif len(ans) > 1 and count==0:
            for z in range(len(ans)-1,-1,-1):
                #print("inside")
                if ans[z]!=0:
                    return ans[z]
                
            return 0
        else:
            return count
        

File: cpp_solutions/58. Length of Last Word.cpp
//ss
class Solution {
public:
    int lengthOfLastWord(string s) {
        
        //cout << int(' ');
        int prev_count = 0;
        int last_seen = 0;
        for(char ch: s){
            if (int(ch)!=int(' ')){
                prev_count+=1;
            }
            else{
                if (prev_count > 0){
                   last_seen = prev_count; 
                }
                
                prev_count = 0;
            }
        }
        if (prev_count!=0){
            return prev_count;
        }
        return last_seen;
        
    }
};
